pkg/contracts/solana/gateway_message_test.go

```diff
@@ -42,8 +46,36 @@ func Test_MsgWhitelistHash(t *testing.T) {
 		wantHashBytes, err := hex.DecodeString(wantHash)
 		require.NoError(t, err)
 
+		// ACT
 		// create new withdraw message
 		hash := contracts.NewMsgWhitelist(whitelistCandidate, whitelistEntry, chainID, nonce).Hash()
-		require.True(t, bytes.Equal(hash[:], wantHashBytes))
+
+		// ASSERT
+		require.EqualValues(t, hash[:], wantHashBytes)
+	})
+}
+
+func Test_MsgWithdrawSPLHash(t *testing.T) {
+	t.Run("should calculate expected hash", func(t *testing.T) {
+		// ARRANGE
+		// #nosec G115 always positive
+		chainID := uint64(chains.SolanaLocalnet.ChainId)
+		nonce := uint64(0)
+		amount := uint64(1336000)
+		mintAccount := solana.MustPublicKeyFromBase58("AS48jKNQsDGkEdDvfwu1QpqjtqbCadrAq9nGXjFmdX3Z")
+		to := solana.MustPublicKeyFromBase58("37yGiHAnLvWZUNVwu9esp74YQFqxU1qHCbABkDvRddUQ")
+		toAta, _, err := solana.FindAssociatedTokenAddress(to, mintAccount)
+		require.NoError(t, err)
+
+		wantHash := "87fa5c0ed757c6e1ea9d8976537eaf7868bc1f1bbf55ab198a01645d664fe0ae"
+		wantHashBytes, err := hex.DecodeString(wantHash)
```

Comment:
You could use `wantHashBytes := testutil.HexToBytes(t, wantHash)` for simplification if you want. Same for other occurrences.